packages/flutter_tools/lib/src/base/error_handling_file_system.dart

```diff
@@ -36,6 +37,16 @@ class ErrorHandlingFileSystem extends ForwardingFileSystem {
 
   @override
   File file(dynamic path) => ErrorHandlingFile(delegate, delegate.file(path));
+
+  @override
+  p.Context get path => _cachedPath ??= delegate.path;
```

Comment:
We should probably have a comment here explaining that this only works because changing the cwd through `dart:io` is forbidden.

Reply:
Do we have a test(can we add a test if not) that asserts we do not set the cwd through dart:io?

Reply:
We don't allow imports of dart:io via the forbidden imports test.